### PR TITLE
.Net: Update ONNX Rag sample to use vector store.

### PR DESCRIPTION
**Pull request by westey-m**
### Motivation and Context

#9377

### Description

Update ONNX Rag sample to use vector store.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

**Comment by RogerBarreto** [comment]
This is an excellent use case for Migration Guidance. Will save this PR as an example BTW :)

File: dotnet/samples/Demos/OnnxSimpleRAG/Program.cs [hunk 1]
```diff
@@ -3,15 +3,18 @@
...
 using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.VectorData;
 using Microsoft.SemanticKernel;
 using Microsoft.SemanticKernel.ChatCompletion;
+using Microsoft.SemanticKernel.Connectors.InMemory;
+using Microsoft.SemanticKernel.Data;
 using Microsoft.SemanticKernel.Embeddings;
-using Microsoft.SemanticKernel.Memory;
-using Microsoft.SemanticKernel.Plugins.Memory;
+using Microsoft.SemanticKernel.PromptTemplates.Handlebars;
```
Comment (RogerBarreto) [inline dotnet/samples/Demos/OnnxSimpleRAG/Program.cs:17]: Not very sure of adding this complexity to the demonstraction. Maybe having the `Simple` and a `HandlerBars` approach?

```suggestion
using Microsoft.SemanticKernel.PromptTemplates.Handlebars;
```

**Review by RogerBarreto — APPROVED** [review]
Would keep it simple (without Handler bars) or add this as an option.

**Review by markwallace-microsoft — APPROVED** [review]

File: dotnet/samples/Demos/OnnxSimpleRAG/Program.cs (hunk 1 above)
Comment (westey-m) [reply dotnet/samples/Demos/OnnxSimpleRAG/Program.cs:17]: Simple isn't working unfortunately.  The search response is returning a list of strings, but they are rendered as
`Answer the question using the memory content: System.Collections.Generic.List1[System.String]`

@markwallace-microsoft is adding a work item to try and get this addressed in the default prompt template engine

**Pull request merged**